lib/wcc/contentful/model_methods.rb

```diff
@@ -81,14 +81,31 @@ def to_h(stack = nil)
     raise WCC::Contentful::CircularReferenceError.new(stack, id) if stack&.include?(id)
 
     stack = [*stack, id]
-
+    typedef = self.class.content_type_definition
     fields =
-      self.class::FIELDS.each_with_object({}) do |field, h|
-        if val = instance_variable_get('@' + field + '_resolved')
-          val = _try_map(val) { |v| v ? v.to_h(stack) : v }
+      typedef.fields.each_with_object({}) do |(name, field_def), h|
+        if field_def.type == :Link || field_def.type == :Asset
+          if _resolved_field?(name, 0)
+            val = public_send(name)
+            val = _try_map(val) { |v| v ? v.to_h(stack) : v }
```

Comment:
need the ternary here? what are values that would be falsey, but we still want to keep?

Reply:
nil, in case someone links to an unpublished entry

Reply:
maybe I should build that into try_map instead